appIconIndicators.js

```diff
@@ -941,14 +944,17 @@ var DominantColorExtractor = class DashToDock_DominantColorExtractor {
         if (iconTexture instanceof Gio.FileIcon) {
             // Use GdkPixBuf to load the pixel buffer from the provided file path
             return GdkPixbuf.Pixbuf.new_from_file(iconTexture.get_file().get_path());
+        } else if (iconTexture instanceof Gio.ThemedIcon) {
+            // Get the pixel buffer from the icon theme
+            let icon_info = themeLoader.lookup_icon(iconTexture.get_names()[0], DOMINANT_COLOR_ICON_SIZE, 0);
+            if (icon_info !== null)
+                return icon_info.load_icon();
+            else
+                return null;
+        } else {
+            // Use GdkPixBuf to load the pixel buffer from memory
+            return GdkPixbuf.Pixbuf.new_from_stream(iconTexture.load(DOMINANT_COLOR_ICON_SIZE, null)[0], null);
```

Comment:
I think it would be possible to do this in an async way, by doing some rerfactor so that the the value is cached.

Reply:
Check https://gjs.guide